exchanges/bybit/bybit.go

```diff
@@ -8,740 +8,2261 @@ import (
+}
+
+// SetTakeProfitStopLossMode set partial TP/SL mode, you can set the TP/SL size smaller than position size.
+func (by *Bybit) SetTakeProfitStopLossMode(ctx context.Context, arg *TPSLModeParams) (*TPSLModeResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	switch arg.Category {
+	case "":
+		return nil, errCategoryNotSet
+	case cLinear, cInverse:
+	default:
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, arg.Category)
+	}
+	switch {
+	case arg.Symbol == "":
+		return nil, errSymbolMissing
+	case arg.TpslMode == "":
+		return nil, errTakeProfitOrStopLossModeMissing
+	}
+	var resp *TPSLModeResponse
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/position/set-tpsl-mode", nil, arg, &resp, setPositionTPLSModeEPL)
+}
+
+// SwitchPositionMode switch the position mode for USDT perpetual and Inverse futures.
+// If you are in one-way Mode, you can only open one position on Buy or Sell side.
+// If you are in hedge mode, you can open both Buy and Sell side positions simultaneously.
+// switches mode between MergedSingle: One-Way Mode or BothSide: Hedge Mode
+func (by *Bybit) SwitchPositionMode(ctx context.Context, arg *SwitchPositionModeParams) error {
+	if arg == nil {
+		return errNilArgument
+	}
+	switch arg.Category {
+	case "":
+		return errCategoryNotSet
+	case cLinear, cInverse:
+	default:
+		return fmt.Errorf("%w, category: %s", errInvalidCategory, arg.Category)
+	}
+	if arg.Symbol.IsEmpty() && arg.Coin.IsEmpty() {
+		return errEitherSymbolOrCoinRequired
+	}
+	return by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/position/switch-mode", nil, arg, &struct{}{}, defaultEPL)
+}
+
+// SetRiskLimit risk limit will limit the maximum position value you can hold under different margin requirements.
+// If you want to hold a bigger position size, you need more margin. This interface can set the risk limit of a single position.
+// If the order exceeds the current risk limit when placing an order, it will be rejected.
+// '0': one-way mode '1': hedge-mode Buy side '2': hedge-mode Sell side
+func (by *Bybit) SetRiskLimit(ctx context.Context, arg *SetRiskLimitParam) (*RiskLimitResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	switch arg.Category {
+	case "":
+		return nil, errCategoryNotSet
+	case cLinear, cInverse:
+	default:
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, arg.Category)
+	}
+	if arg.PositionMode < 0 || arg.PositionMode > 2 {
+		return nil, errInvalidPositionMode
+	}
+	if arg.Symbol.IsEmpty() {
+		return nil, errSymbolMissing
+	}
+	var resp *RiskLimitResponse
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/position/set-risk-limit", nil, arg, &resp, setPositionRiskLimitEPL)
+}
+
+// SetTradingStop set the take profit, stop loss or trailing stop for the position.
+func (by *Bybit) SetTradingStop(ctx context.Context, arg *TradingStopParams) error {
```

Comment:
The `TradingStopParams` args could use some `float64`s and the `json` tag `,string`

Reply:
I left a comment regarding the effect of value 0.